### PR TITLE
Jsweek7

### PR DESCRIPTION
**Pull request by lalitchoudhary09**
7.1 Reverse Link List
7 1 Reverse Link List
7.2 RotateLinkList
7 2 RotateLinkList
7.3 Detect Loop
7 3 Detect Loop
7.4 Check Parenthesis
7 4 Check Parenthesis
7.5 Next Greater Element
7 5 Next Greater Element
7.6 Queue Using Stack
7 6 Queue Using Stack

@anilrao316

### CODE REVIEW

File: Week-7/Problem 7.2 AS RotateLinkedList.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    {
+       
+        this.value=value;
+        this.next=null;
+    }
+}
+
+function createSLL(arr)
+{
+    let node = new SLL(arr[0]);
```
Comment (anilrao316) [inline Week-7/Problem 7.2 AS RotateLinkedList.js]: can be const

File: Week-7/Problem 7.4 Check Parenthesis.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+console.log("Problem 7.4: Check Parenthesis");
+console.log("Time complexity O(n) and space complexity O(n)");
+class Stack
```
Comment (anilrao316) [inline Week-7/Problem 7.4 Check Parenthesis.js]: this can go in as seperate file

**Comment by anilrao316** [comment]
rest looks good

File: Week-7/Problem 7.2 AS RotateLinkedList.js (hunk 1 above)
Comment (lalitchoudhary09) [reply Week-7/Problem 7.2 AS RotateLinkedList.js]: Crated a sperate file for classes and handle there.

File: Week-7/Problem 7.4 Check Parenthesis.js (hunk 2 above)
Comment (lalitchoudhary09) [reply Week-7/Problem 7.4 Check Parenthesis.js]: Crated a sperate file for classes. 
classes.js

**Pull request merged**